src/repository.rs

```diff
@@ -317,11 +321,18 @@ where
         HttpRepository {
             url: url,
             client: client,
+            http_hook: None,
             user_agent: user_agent,
             _interchange: PhantomData,
         }
     }
 
+    /// Set an hook function to be invoked before sending an HTTP request.
+    pub fn set_http_hook(&mut self, hook: Option<HttpHook>) -> &Self {
```

Comment:
Pedantic: I'm trying to avoid the Java `set_xxx` convention, so could you change this to just `http_hook`.

Less pedantic: Could you put a short example in the docs showing how someone might use this to set auth headers (or whatever your use case was). It possible the description isn't enough.